node/builder_chain.go

```diff
@@ -280,14 +277,15 @@ func ConfigFullNode(c interface{}) Option {
 			),
 		),
 
-		// enable message index for full node when configured by the user, otherwise use dummy.
-		If(cfg.Index.EnableMsgIndex, Override(new(index.MsgIndex), modules.MsgIndex)),
-		If(!cfg.Index.EnableMsgIndex, Override(new(index.MsgIndex), modules.DummyMsgIndex)),
-
 		// enable fault reporter when configured by the user
 		If(cfg.FaultReporter.EnableConsensusFaultReporter,
 			Override(ConsensusReporterKey, modules.RunConsensusFaultReporter(cfg.FaultReporter)),
 		),
+
+		Override(new(index.Indexer), modules.ChainIndexer(cfg.ChainIndexer)),
+		If(!cfg.ChainIndexer.DisableIndexer,
+			Override(InitChainIndexerKey, modules.InitChainIndexer),
+		),
```

Comment:
I think we want to move this up into an `ApplyIf(isFullNode` or make a new one here; I don't think it makes any sense to run this on a lite node, it'd be quite expensive to do it all over RPC, we'd end up with a database file, which would be surprising I think, and it's not going to buy us much if the upstream RPC endpoint is doing it anyway.

Reply:
@rvagg Done.